articles/android-nested-recycler-view/index.md

```diff
@@ -0,0 +1,400 @@
+            arrayList.add(new ChildModel( R.drawable.bestofoscar1,"Movie Name"));
+        }
+
+        // added in fifth child row
+        if (parentModelArrayList.get(position).movieCategory().equals("Category5")) {
+            arrayList.add(new ChildModel( R.drawable.moviedubbedinhindi4,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.hollywood2,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.bestofoscar4,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.mov2,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.hollywood1,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.bestofoscar1,"Movie Name"));
+        }
+
+        // added in sixth child row
+        if (parentModelArrayList.get(position).movieCategory().equals("Category6")) {
+            arrayList.add(new ChildModel(R.drawable.hollywood5,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.blackp,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.bestofoscar4,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.moviedubbedinhindi6,"Movie Name"));
+            arrayList.add(new ChildModel( R.drawable.hollywood1,"Movie Name"));
+            arrayList.add(new ChildModel(R.drawable.bestofoscar6,"Movie Name"));
+        }
+
+        ChildRecyclerViewAdapter childRecyclerViewAdapter = new ChildRecyclerViewAdapter(arrayList,holder.childRecyclerView.getContext());
+            holder.childRecyclerView.setAdapter(childRecyclerViewAdapter);
+        }
+}
+```
+
+As you can see, we are binding the information of the  `ChildRecyclerViewAdapter` to the `ParentRecyclerViewAdapter`. I have set different data categories. In this case, each category contains its distinct data. The `ArrayList` of all the categories binds to the parent RecyclerView. In the `onBindViewHolder`, we are setting up the child RecyclerView by
+
+- Initializing and setting the [`LinearLayoutManager`](https://developer.android.com/reference/androidx/recyclerview/widget/LinearLayoutManager) (create an instance of the layout manager) for the child RecyclerView. Here we specify that this view will be `HORIZONTAL`.
+- Invoke the layout fixed size. This tells the layout not to move around the screen. It stabilizes the layout screen movements.
+- Pass `ArrayList` of the child items to the respective `ParentViewHolder` as `category.setText(currentItem.movieCategory())` of the parent RecyclerView. Each category will inflate with respective array list items consisting of a movie name and a hero image. Make sure to check the content of this project on [GitHub](https://github.com/kimkimani/Nested_RecyclerView) to get all the necessary image resources for each `movieCategory`.
+- Associate the child Adapter and the RecyclerView. This informs the RecyclerView which adapter it should work with.
+
+Let set the necessary categories to display the movies to the main RecyclerView layout.
```

Comment:
Consider rewriting "Let set the necessary ..." to "Let us set the necessary ..."